src/GraphQL.Tests/Types/Collections/SchemaTypesTests.cs

```diff
@@ -45,6 +46,46 @@ public void does_not_request_instance_more_than_once()
         mock.VerifyNoOtherCalls();
     }
 
+    [Fact]
+    public void throws_exception_when_multiple_type_instances_exists()
+    {
+        var schema = new Schema
+        {
+            NameConverter = new CamelCaseNameConverter()
+        };
+
+        var queryGraphType = new ObjectGraphType
+        {
+            Name = "Query"
+        };
+
+        schema.RegisterType(queryGraphType);
+
+        // Object 1
+        var graphType1 = new ObjectGraphType
+        {
+            Name = "MyObject"
+        };
+
+        graphType1.Field<IntGraphType>("int");
+
+        queryGraphType.Field("first", graphType1);
+
+        // Object 2
+        var graphType2 = new ObjectGraphType
+        {
+            Name = "MyObject"
+        };
+
+        graphType2.Field<IntGraphType>("int");
+        graphType2.Field<StringGraphType>("string");
+
+        queryGraphType.Field("second", graphType2);
+
+        // Test
+        Assert.Throws<InvalidOperationException>(() => schema.Initialize());
```

Comment:
We use the Shouldly assertion library for tests, and as a matter of convention, test the message name (more so it can be seen what the failure is rather than because we want to ensure it does not change).

```suggestion
        Should.Throw<InvalidOperationException>(() => schema.Initialize())
            .Message.ShouldBe("...");
```